File: Searching_algorithms/C/Linear_Search.c
// Time:
// Best case: O(1)
// Average case: O(n)
// Worst case: O(n)

// Space: O(1)

int linearSearch(int* arr, int key, int len) {
  
  for(int idx = 0; idx < len; idx++) {
    if(key == arr[idx]) {
      return idx;
    }
  }
  
  return -1;
}

File: Searching_algorithms/Python/Linear_Search.py
# Time:
# Best case: O(1)
# Average case: O(n)
# Worst case: O(n)

# Space: O(1)

def linearSearch(arr, key, n):

  for idx in range(n):
    if key == arr[idx]:
      return idx
      
  return -1
